packages/ed25519-signature-2018/src/Ed25519Signature2018.ts

```diff
@@ -28,7 +28,12 @@ export class Ed25519Signature2018 {
   public verificationMethod?: string;
   constructor(options: IEd25519Signature2018Options = {}) {
     this.signer = options.signer;
-    this.date = options.date;
+    if (options.date) {
+      this.date = new Date(options.date);
```

Comment:
this converts the date from what is supplied in the option.... if we are doing this for compatibility, we should warn... anything that changes an input provided by the caller should be a warning, or an error.